include/envoy/http/codec.h

```diff
@@ -189,6 +210,9 @@ struct Http2Settings {
   // our default connection-level window also equals to our stream-level
   static const uint32_t DEFAULT_INITIAL_CONNECTION_WINDOW_SIZE = 256 * 1024 * 1024;
   static const uint32_t MAX_INITIAL_CONNECTION_WINDOW_SIZE = (1U << 31) - 1;
+
+  // By default, do not enforce stream buffer limits beyond the connection level limits.
+  static const uint32_t DEFAULT_PER_STREAM_BUFFER_LIMIT = 0;
```

Comment:
Oh, you wanted to remove the option entirely and just go with window size?  Hm... I could imagine a use case where a user wanted to keep the stream limits smaller to avoid one stream hogging full connection bandwidth but was willing to buffer locally to spool upstream immediately as window became available.  I can comment up the reason they're somewhat distinct use cases, or just remove it until/if someone asks for it back.